code/modules/projectiles/gun.dm

```diff
@@ -63,6 +63,7 @@
 	var/scoped_accuracy = null
 	var/list/burst_accuracy = list(0) //allows for different accuracies for each shot in a burst. Applied on top of accuracy
 	var/list/dispersion = list(0)
+	var/is_charging = 0 //special var for sanity checks in the three guns that currently use charging as a special_check

```

Comment:
I don't see a point in defining this variable this high up. Defining it for the `/obj/item/weapon/gun/energy/vaurca` class would be better.

Variables should be defined for as small a scopes as humanly possible.